compiler/AST/symbol.cpp

```diff
@@ -1028,11 +1028,28 @@ bool ShadowVarSymbol::isConstant() const {
 }
 
 bool ShadowVarSymbol::isConstValWillNotChange() {
-  //
-  // This is written to only be called post resolveIntents
-  //
-  INT_ASSERT(intent != TFI_DEFAULT && intent != TFI_CONST);
-  return intent == TFI_CONST_IN;
+  switch (intent) {
+    case TFI_DEFAULT:
+    case TFI_CONST:
+      // Caller responsibility - no abstract intents please.
+      INT_ASSERT(false);
+      return false;
+
+    case TFI_CONST_IN:
+    case TFI_IN_OUTERVAR: // should these two be here?
```

Comment:
If "Should these two be here" is a question for me, I don't know, but I suppose it's probably OK to consider them const...